core/random.rbs

```diff
@@ -71,6 +91,7 @@ class Random < RBS::Unnamed::Random_Base
   def self.rand: () -> Float
                | (Integer | ::Range[Integer] max) -> Integer
                | (Float | ::Range[Float] max) -> Float
+               | (::Range[T]) -> T
```

Comment:
> Yeah, the type we want would be [T <: Numeric] (Range[T]) -> T, but RBS currently doesn't allow writing it.
>
> I think there are two options. [T] (Range[T]) -> T or (Range[Numeric]) -> Numeric. The first option would make sense in my opinion, because testing the code will uncover the problem anyway.
> https://github.com/ruby/rbs/pull/470#discussion_r521191243

Is `[T] (Range[T])` invalid syntax?

Reply:
`[T] (Range[T])` is not a valid piece of code. It becomes a valid method type by adding the returned type, like `[T] (Range[T]) -> T` as soutaro said.

By the way, `[T < Numeric] (Range[T]) -> T` is more appropriate because the upper bound type for a type parameter has been implemented.